inginious/frontend/pages/tasks.py

```diff
@@ -170,10 +170,9 @@ def GET(self, courseid, taskid, isLTI):
 
             submissions = self.submission_manager.get_user_submissions(task) if self.user_manager.session_logged_in() else []
             user_info = self.database.users.find_one({"username": username})
-
-            # Display the task itself
-            return self.template_helper.get_renderer().task(user_info, course, task, submissions,
-                                                            students, eval_submission, user_task, previous_taskid, next_taskid, self.webterm_link, random_input_list)
+      
+            # Call to adaptive hook
+            return self.plugin_manager.call_hook("adaptive_get_hook", username=username, page=self, course=course, courseid=courseid, task=task, taskid=taskid, students=students, eval_submission=eval_submission, user_task=user_task, user_info=user_info, submissions=submissions, previous_taskid=previous_taskid, next_taskid=next_taskid, random_input_list=random_input_list)[0]
```

Comment:
What If there is no plugin registering "adaptive_get_hook" ? There is nothing displayed?
Moreover, the name of the hook is bad; something like "custom_task_renderer" would be a bit better.

Still, I don't see why you would do that.